### PR TITLE
workflows: turn on linting using `ruff`

### PR DESCRIPTION
**Pull request by andlaus**
This hopefully catches a few bugs before they bite.

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -55,14 +55,18 @@ jobs:
...
-      - name: Static type checking with mypy
+      - name: Static type checking (mypy)
         run: |
           # remove '# type:' comments in version.py which mypy-3.8
           # stumbles over. for some reason, mypy cannot be
           # instructed to ignore 'version.py'
           sed -i "s/# type:.*//" odxtools/version.py
           python -m mypy .
 
+      - name: Lint code quality (ruff)
```
Comment (kayoub5) [inline .github/workflows/tests.yml:66]: What happens if ruff and yapf don't agree?

File: odxtools/cli/browse.py [hunk 2]
```diff
@@ -194,31 +197,36 @@ def encode_message_from_string_values(
 
     # Request values for parameters
     for parameter_sn, parameter_value in parameter_values.items():
+        parameter = param_dict.get(parameter_sn)
+        if parameter is None:
+            print(f"I don't know the parameter {parameter_sn}")
```
Comment (kayoub5) [inline odxtools/cli/browse.py:202]: Don't use print, use warning/logger or odxassert

File: tests/test_encoding.py [hunk 3]
```diff
@@ -302,23 +302,23 @@ def _create_request(self, parameters: List[Parameter]) -> Request:
...
-            base_data_type=DataType.A_UINT32,
-            base_type_encoding=None,
-            is_highlow_byte_order_raw=None,
-            bit_mask=None,
-            is_condensed_raw=None,
-        )
+        unit_kwargs = {
+            "base_data_type": DataType.A_UINT32,
+            "base_type_encoding": None,
+            "is_highlow_byte_order_raw": None,
```
Comment (kayoub5) [inline tests/test_encoding.py:308]: Why?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (andlaus) [reply .github/workflows/tests.yml:66]: a lot of fun ;).

If the disagreement cannot be fixed by "normal" means, the ruff check can be turned off globally in `pyproject.toml` (while doing this PR, this happened with the sorting of the import statement), or ruff can be instructed to ignore a certain error at a given location using `# noqa: $ERRNO` comments (there are some of these in this PR as well...)

**Review by kayoub5 — APPROVED** [review]

**Pull request merged**